bundles/org.openhab.binding.pilight/src/main/resources/OH-INF/thing/devices.xml

```diff
@@ -0,0 +1,118 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="pilight"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="https://openhab.org/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
+
+	<thing-type id="switch">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="bridge"/>
+		</supported-bridge-type-refs>
+
+		<label>Pilight Switch</label>
+		<description>Pilight Switch</description>
+
+		<channels>
+			<channel id="state" typeId="switch-state"/>
+		</channels>
+
+		<config-description-ref uri="thing-type:pilight:device"/>
+	</thing-type>
+
+	<thing-type id="contact">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="bridge"/>
+		</supported-bridge-type-refs>
+
+		<label>Pilight Contact</label>
+		<description>Pilight Contact</description>
+
+		<channels>
+			<channel id="state" typeId="contact-state"/>
+		</channels>
+
+		<config-description-ref uri="thing-type:pilight:device"/>
+	</thing-type>
+
+	<thing-type id="dimmer">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="bridge"/>
+		</supported-bridge-type-refs>
+
+		<label>Pilight Dimmer</label>
+		<description>Pilight Dimmer</description>
+
+		<channels>
+			<channel id="dimlevel" typeId="dimlevel"/>
+		</channels>
+
+		<config-description-ref uri="thing-type:pilight:device"/>
+	</thing-type>
+
+	<thing-type id="generic" extensible="string,number">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="bridge"/>
+		</supported-bridge-type-refs>
+
+		<label>Pilight Generic Device</label>
+		<description>Pilight Generic Device</description>
+
+		<config-description-ref uri="thing-type:pilight:device"/>
+	</thing-type>
+
+	<channel-type id="contact-state">
+		<item-type>Contact</item-type>
+		<label>State of contact</label>
+		<description>State of pilight contact.</description>
+		<state readOnly="true"/>
+	</channel-type>
+
+	<channel-type id="dimlevel">
```

Comment:
There are default channel types defined. You could use `brightness` here instead of defining your own. See https://www.openhab.org/docs/developer/bindings/thing-xml.html#system-state-channel-types

Same for the switch.